### PR TITLE
Create namespace to run ci and integrate pipelines-as-code

### PR DESCRIPTION
**Pull request by joejstuart**
This PR creates a namespace and integrates pipelines-as-code to run ci for the infra-deployments repo

### CODE REVIEW

File: argo-cd-apps/overlays/development/repo-overlay.yaml [hunk 1]
```diff
@@ -41,3 +41,14 @@ spec:
   source:
     repoURL: https://github.com/redhat-appstudio/infra-deployments.git
     targetRevision: main
+---
+apiVersion: argoproj.io/v1alpha1
```
Comment (sbose78) [inline argo-cd-apps/overlays/development/repo-overlay.yaml:45]: Wondering how it's going to be used/tested in development mode?

Comment (joejstuart) [reply argo-cd-apps/overlays/development/repo-overlay.yaml]: I just had that there to test with my fork. I guess it's not needed.

Comment (sbose78) [reply argo-cd-apps/overlays/development/repo-overlay.yaml]: No it's fine, if you've found it useful, I'd be curious how you've used it, then we could have devs use this as well.

Comment (joejstuart) [reply argo-cd-apps/overlays/development/repo-overlay.yaml]: It was useful to test the pipeline runs. I submitted a PR to my fork as the test case. The only pain point was adding the GitHub app secret, app id and route on each deployment. I can create a script to make that easier though.

File: .ci/pipelines-as-code/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+resources:
+- pipelines-as-code.yaml
+- pvc.yaml
+- tasks/tekton-chains-tasks.yaml
```
[the remark this answers is not in the document]
Comment (joejstuart) [reply .ci/pipelines-as-code/kustomization.yaml:4]: It's a task that pipelines as code will run eventually.

File: .ci/pipelines-as-code/tasks/tekton-chains-tasks.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  params:
+    - name: task-url
+      type: string
+      description: The url to a task to run
+      default:  https://raw.githubusercontent.com/tektoncd/chains/main/examples/taskruns/task-output-image.yaml
+  workspaces:
+    - name: source
+  steps:
+    - name: cosign-taskrun
+      image: quay.io/redhat-appstudio/appstudio-utils:latest
```
[the remark this answers is not in the document]
Comment (joejstuart) [reply .ci/pipelines-as-code/tasks/tekton-chains-tasks.yaml]: fixed

File: argo-cd-apps/overlays/development/repo-overlay.yaml (hunk 1 above)
Comment (joejstuart) [reply argo-cd-apps/overlays/development/repo-overlay.yaml:45]: Hi @sbose78 here's my thoughts on testing it in dev mode. The commit message has a description of the usage. https://github.com/redhat-appstudio/infra-deployments/pull/126/commits/4ef258b694d283b5a9b5b4c28d889da0faf466b8

**Pull request closed without merging**